### PR TITLE
Feature/infinite scroll

### LINKED ISSUE #366: Pagination of Posts in feeds page
### Description 

- Posts in the feed page  should lazily load more data as you scroll down or while clicking on a button for example `load more`
( you need to choose the easiest one ) and let me know your choice before you start.

### Backed requirements (implmentation in progress) 
-  Implement the pagination and share the strategy with us to make the required changes.
- please share with us the required changes from the `front-end` so that we can work in parallel

### PR DESCRIPTION
**Pull request by atwfk**
closes #366

### feature required
apply infinite scroll and load more posts when user browsing posts feed

## files changed
- install react-infinite-scroll-component package
- use **InfiniteScroll** component and created the `getMorePosts` handler
- add postsLimit to the configuration file

### CODE REVIEW

File: modules/HomePage/components/Posts/Posts.tsx [hunk 1]
```diff
@@ -31,6 +40,35 @@ const Posts: FC<IPostFeed.IPosts> = ({ data }): ReactElement => {
     return null;
   };
 
+  const getMorePosts = async (): Promise<void> => {
+    const loggedInUser = getUserToken();
+    if (!loggedInUser) {
```
Comment (AhmedAdel999) [inline modules/HomePage/components/Posts/Posts.tsx:45]: If the user isn't logged in he will be redirected to login page before even reaching the client why the need to check again?

Comment (atwfk) [reply modules/HomePage/components/Posts/Posts.tsx:45]: I'm checking if the user is logged in but the token is expired, then no need to make an API call

File: modules/HomePage/components/Posts/Posts.tsx [hunk 2]
```diff
@@ -10,14 +10,23 @@ import { EStatusCode } from '@modules/shared/api/EStatusCode';
...
 
 const Posts: FC<IPostFeed.IPosts> = ({ data }): ReactElement => {
   const [posts, setPosts] = useState<IPostFeed.IPost[]>(data.posts);
+  const [hasMore, setHasmore] = useState<boolean>(true);
+  const [postsCount, setPostsCount] = useState<number>(data.postsCount);
   const { redirectToLoginPage } = useRedirect();
   const toastId = useRef<ReactText>();
 
+  useEffect(() => {
+    setHasmore(posts.length < postsCount + 100);
```
Comment (AhmedAdel999) [inline modules/HomePage/components/Posts/Posts.tsx:27]: Why `postsCount + 100` here?

File: modules/HomePage/components/Posts/Posts.tsx (hunk 1 above)
Comment (AhmedAdel999) [reply modules/HomePage/components/Posts/Posts.tsx:45]: So of the token is expired the cookie will be automatically deleted?

File: modules/HomePage/components/Posts/Posts.tsx (hunk 2 above)
Comment (atwfk) [reply modules/HomePage/components/Posts/Posts.tsx:27]: to ensure that the feature works fine until the BE implement the limit and the offset

**Review by AhmedAdel999 — APPROVED** [review]
LGTM

**Pull request merged**